tests/python/pants_test/backend/python/tasks/test_python_distribution_integration.py

```diff
@@ -179,25 +178,26 @@ def test_pants_resolves_local_dists_for_current_platform_only(self):
 
   def _get_current_platform_string(self):
     return Platform.create().resolve_platform_specific({
+      # TODO: this will fail on everyone's laptop if they're not at 10.12.
       'darwin': lambda: 'macosx-10.12-x86_64',
       'linux': lambda: 'linux-x86_64',
     })
 
-  def test_pants_tests_local_dists_for_current_platform_only(self):
-    platform_string = self._get_current_platform_string()
-    # Use a platform-specific string for testing because the test goal
-    # requires the coverage package and the pex resolver raises an Untranslatable error when
-    # attempting to translate the coverage sdist for incompatible platforms.
-    pants_ini_config = {'python-setup': {'platforms': [platform_string]}}
-    # Clean all to rebuild requirements pex.
-    # TODO: why do we need to clean-all here?
+  def test_pants_tests_local_dists_for_simulated_current_platform_only(self):
+    """
+    The "simulated" current platform here is the platform that 'current' will resolve to on this
+    host -- we do support using those platform strings directly as well.
+    """
     with temporary_dir() as tmp_dir:
       command=[
         '--pants-distdir={}'.format(tmp_dir),
-        'clean-all',
         'test',
         '{}:fasthello'.format(self.fasthello_tests)]
-      pants_run = self.run_pants(command=command, config=pants_ini_config)
+      pants_run = self.run_pants(command=command, config={
+        'python-setup': {
+          'platforms': [self._get_current_platform_string()],
```

Comment:
Why not use 'current'? That shortcut has existed >5 years.

Reply:
I added this + explanation because I wasn't sure why the platform string method existed. Removed in 4456105.